services/license-store.test.js

```diff
@@ -0,0 +1,179 @@
+      expect(license.name).toBe('CC BY 3.0 HK');
+    });
+
+    it('detects ported "CC BY-SA 2.0"', () => {
+      const license = subject.match(['Cc-by-sa-2.0-uk']);
+      expect(license.id).toBe('cc-by-sa-2.0-ported');
+      expect(license.name).toBe('CC BY-SA 2.0 UK');
+    });
+
+    it('detects ported "CC BY-SA 2.5"', () => {
+      const license = subject.match(['Cc-by-sa-2.5-it']);
+      expect(license.id).toBe('cc-by-sa-2.5-ported');
+      expect(license.name).toBe('CC BY-SA 2.5 IT');
+    });
+
+    it('detects ported "CC BY-SA 3.0"', () => {
+      const license = subject.match(['Cc-by-sa-3.0-tw']);
+      expect(license.id).toBe('cc-by-sa-3.0-ported');
+      expect(license.name).toBe('CC BY-SA 3.0 TW');
+    });
+  });
+
+  describe('compatible()', () => {
+    it('finds no compatible licences for "cc by-sa 4.0"', () => {
+      expect(subject.compatible('cc-by-sa-4.0')).toEqual([]);
+    });
+
+    const cases = {
+      'cc-by-sa-3.0': ['cc-by-sa-3.0-de', 'cc-by-sa-4.0'],
+      'cc-by-sa-3.0-de': ['cc-by-sa-3.0', 'cc-by-sa-4.0'],
+      'cc-by-sa-2.5': ['cc-by-sa-3.0-de', 'cc-by-sa-3.0', 'cc-by-sa-4.0'],
+      'cc-by-sa-2.0': [
+        'cc-by-sa-2.0-de',
+        'cc-by-sa-2.5',
+        'cc-by-sa-3.0-de',
+        'cc-by-sa-3.0',
+        'cc-by-sa-4.0',
+      ],
+      'cc-by-sa-2.0-de': [
+        'cc-by-sa-2.0',
+        'cc-by-sa-2.5',
+        'cc-by-sa-3.0-de',
+        'cc-by-sa-3.0',
+        'cc-by-sa-4.0',
+      ],
+      'cc-by-sa-1.0': [
+        'cc-by-sa-2.0',
+        'cc-by-sa-2.0-de',
+        'cc-by-sa-2.5',
+        'cc-by-sa-3.0-de',
+        'cc-by-sa-3.0',
+        'cc-by-sa-4.0',
+      ],
+      'cc-by-4.0': ['cc-by-sa-4.0'],
+      'cc-by-3.0': ['cc-by-3.0-de', 'cc-by-sa-3.0-de', 'cc-by-sa-3.0'],
+      'cc-by-3.0-de': ['cc-by-3.0', 'cc-by-sa-3.0-de', 'cc-by-sa-3.0'],
+      'cc-by-2.5': [
+        'cc-by-3.0-de',
+        'cc-by-3.0',
+        'cc-by-4.0',
+        'cc-by-sa-2.5',
+        'cc-by-sa-3.0-de',
+        'cc-by-sa-3.0',
+      ],
+      'cc-by-2.0': [
+        'cc-by-2.0-de',
+        'cc-by-2.5',
+        'cc-by-3.0-de',
+        'cc-by-3.0',
+        'cc-by-4.0',
+        'cc-by-sa-2.0-de',
+        'cc-by-sa-2.0',
+      ],
+      'cc-by-2.0-de': [
+        'cc-by-2.0',
+        'cc-by-2.5',
+        'cc-by-3.0-de',
+        'cc-by-3.0',
+        'cc-by-sa-2.0-de',
+        'cc-by-sa-2.0',
+      ],
+      'cc-by-1.0': [
+        'cc-by-2.0-de',
+        'cc-by-2.0',
+        'cc-by-2.5',
+        'cc-by-3.0-de',
+        'cc-by-3.0',
+        'cc-by-sa-1.0',
+      ],
+    };
+
+    Object.keys(cases).forEach(id => {
+      const expected = cases[id];
+
+      it(`finds compatible licenses for "${id}"`, () => {
+        const result = subject.compatible(id);
+        expect(result).toHaveLength(expected.length);
+        const ids = result.map(license => license.id);
+        expect(ids).toEqual(expect.arrayContaining(expected));
+      });
+    });
+  });
+});
```

Comment:
For comparison: https://github.com/wmde/Lizenzhinweisgenerator/blob/0df1805ef1b362b56c287b7035b2bbf66e6642a1/tests/app/LicenceStore.tests.js